src/Traces.Web/ViewModels/TracesBaseViewModel.cs

```diff
@@ -193,7 +193,11 @@ protected void AddTraceToDictionary(TraceItemModel trace)
         {
             if (SortedGroupedTracesDictionary.ContainsKey(trace.DueDate))
             {
-                SortedGroupedTracesDictionary[trace.DueDate].Add(trace);
+                var list = SortedGroupedTracesDictionary[trace.DueDate];
+                if (list.TrueForAll(item => item.Id != trace.Id))
```

Comment:
```suggestion
                if (list.Exists(item => item.Id == trace.Id))
```
and if it exists just `return` and then just add an `else` where you add the item to the list.

I believe having the `Exists` makes it easier to read the code.